credoai/evaluators/model_profiler.py

```diff
@@ -0,0 +1,202 @@
+from typing import Optional
+
+from pandas import DataFrame
+
+from credoai.evaluators import Evaluator
+from credoai.evaluators.utils.validation import check_existence
+from credoai.evidence.containers import ModelProfilerContainer
+from credoai.utils import ValidationError, global_logger
+
+USER_INFO_TEMPLATE = {
+    "developed_by": None,
+    "shared_by": None,
+    "model_type": None,
+    "intended_use": None,
+    "downstream_use": None,
+    "out_of_scope_use": None,
+    "language": None,
+    "related_models": None,
+    "license": None,
+    "resources_for_more_info": None,
+    "input_description": None,
+    "output_description": None,
+    "performance_evaluated_on": None,
+    "limitations": None,
+}
+
+PROTECTED_KEYS = [
+    "model_name",
+    "python_model_type",
+    "library",
+    "model_architecture",
+    "feature_names",
+]
+
+
+class ModelProfiler(Evaluator):
+    """
+    Model profiling evaluator.
+
+    This evaluator builds a model card the purpose of which is to characterize
+    a fitted model.
+
+    The overall strategy is:
+        1. Extract all potentially useful info from the model itself in an
+            automatic fashion.
+        2. Allow the user to personalize the model card freely.
+
+    The method generate_template() provides a dictionary with several entries the
+    user could be interested in filling up.
+
+    Parameters
+    ----------
+    model_info : Optional[dict], optional
+        Information provided by the user that cannot be inferred by
+        the model itself. The dictionary con contain any number of elements,
+        a template can be provided by running the generate_template() method.
```

Comment:
Worried about cases like: the user specifies model_type as LogisticRegression and the inferred model type is a GLM (with hyperparameters such that it really is just logistic regression, or logit with a link function). The info in the model card would be at varying levels of precision and both would be displayed, which might be confusing to someone reading the report.

Reply:
Yeah I think I can expand the protected keys list, so that there is no overwriting of automatically generated fields.
For other confusing info... not sure there is an immediate solution, we can probably be clear on the which info is user generated and which
automatically picked up to begin with.